packages/toolpad-app/src/toolpad/AppEditor/PageEditor/QueryEditor/index.tsx

```diff
@@ -130,8 +130,12 @@ export default function QueryEditor() {
   const { queries = [] } = appDom.getChildNodes(dom, page) ?? [];
 
   const handleEditStateDialogClose = React.useCallback(() => {
-    appStateApi.setView({ kind: 'page', nodeId: page.id });
-  }, [appStateApi, page.id]);
+    if (!dialogState?.isDraft) {
```

Comment:
Prefer positive logic
```suggestion
    if (dialogState?.isDraft) {
      setDialogState(null);
      // ...
```